EnergyPlus_Engine/Create/GroundTemperatures.cs

```diff
@@ -0,0 +1,135 @@
+using System.Linq;
+using BH.oM.Adapters.EnergyPlus;
+using BH.oM.Reflection;
+using BH.oM.Reflection.Attributes;
+
+namespace BH.Engine.Adapters.EnergyPlus
+{
+    public static partial class Create
+    {
+        [Description("Create EnergyPlus SiteGroundTemperatureBuildingSurface, SiteGroundTemperatureShallow and SiteGroundTemperatureDeep objects from an EPW file")]
+        [Input("epwFile", "Path to an epw-file")]
+        [MultiOutput(0, "siteGroundTemperatureBuildingSurface", "An energyPlus SiteGroundTemperatureBuildingSurface object")]
+        [MultiOutput(1, "siteGroundTemperatureBuildingShallow", "An energyPlus SiteGroundTemperatureShallow object")]
+        [MultiOutput(2, "siteGroundTemperatureBuildingDeep", "An energyPlus SiteGroundTemperatureDeep object")]
+        public static Output<SiteGroundTemperatureBuildingSurface, SiteGroundTemperatureShallow, SiteGroundTemperatureDeep> GroundTemperature(string epwFile)
+        {
+            // Check that file passed exists, and exit if it doesn't while passing a record of your failure to the Error Log
+            if (!File.Exists(epwFile))
+            {
+                BH.Engine.Reflection.Compute.RecordError("File not found. Did you give the correct path?");
+                return null;
+            }
+            SiteGroundTemperatureBuildingSurface siteGroundTemperatureBuildingSurface = new SiteGroundTemperatureBuildingSurface();
+            SiteGroundTemperatureShallow siteGroundTemperatureShallow = new SiteGroundTemperatureShallow();
+            SiteGroundTemperatureDeep siteGroundTemperatureDeep = new SiteGroundTemperatureDeep();
+            Output<SiteGroundTemperatureBuildingSurface, SiteGroundTemperatureShallow, SiteGroundTemperatureDeep> output = new Output<SiteGroundTemperatureBuildingSurface, SiteGroundTemperatureShallow, SiteGroundTemperatureDeep>
+            {
+                Item1 = siteGroundTemperatureBuildingSurface,
+                Item2 = siteGroundTemperatureShallow,
+                Item3 = siteGroundTemperatureDeep,
+            };
+            string[] EpwFile = System.IO.File.ReadAllLines(epwFile);
+            // Check to see if the file passed contains the requisite number of lines to parse as an EPW (> 8760)
+            if (EpwFile.Count() < 8760)
+            {
+                BH.Engine.Reflection.Compute.RecordError("The file passed has less than 8760 lines. Are you sure it's an EPW?");
+                return null;
+            }
+            string groundTemperatures = EpwFile[3];
+            List<string> commaPoints = groundTemperatures.Split(',').ToList();
+            if (commaPoints.Count() < 17)
+            {
+                BH.Engine.Reflection.Compute.RecordError("The file has less than 12 temperatures. Does the file contain Surface ground temperatures?");
+                return null;
+            }
+            siteGroundTemperatureBuildingSurface.JanuaryGroundTemperature = System.Convert.ToDouble(commaPoints[6]);
```

Comment:
This is applicable to all the objects being created, but it would be good to have some logic checking how many ground temperatures there are, and the depths those temperatures represent.

Here is an example string containing the kind of data being processed:
`GROUND TEMPERATURES,3,.5,,,,29.01,29.16,28.82,28.31,27.01,26.02,25.37,25.19,25.56,26.36,27.40,28.35,2,,,,28.36,28.66,28.57,28.30,27.42,26.64,26.02,25.70,25.80,26.25,26.97,27.73,4,,,,27.81,28.12,28.18,28.08,27.57,27.04,26.57,26.24,26.17,26.38,26.81,27.32
`
If split on ",", index 1 represents the number of depths listed, index 2 is the first depth, index 18 is the second and index 34 is the third depth.

We can then make sure that the objects being created are suitable - anything less than 1m deep would be `SiteGroundTemperatureBuildingSurface`, 1-5m would be suitable for `SiteGroundTemperatureShallow` and greater than that would be `SiteGroundTemperatureDeep`.